Foundation/Operation.swift

```diff
@@ -439,13 +433,16 @@ open class OperationQueue: NSObject {
             }
 
             let block = DispatchWorkItem(flags: .enforceQoS) { () -> Void in
-                if let sema = self._concurrencyGate {
-                    sema.wait()
-                    self._runOperation()
-                    sema.signal()
+                let op = self._dequeueReadyOperation()
```

Comment:
this really isnt the way it works internally to NSOQ; In my spare time I should probably re-draft up a non-KVO variation again that avoids the semaphore. Since the Objective-C implementation does not use a semaphore and if there are no external observers it does not use KVO (it is lazy in that manner). And I think that might be transliteratable to Swift.